pkg/trace/traceutil/trace.go

```diff
@@ -109,24 +109,32 @@ func ChildrenMap(t pb.Trace) map[uint64][]*pb.Span {
 //
 // A span is considered top-level if:
 // - it's a root span
-// - its parent is unknown (other part of the code, distributed trace)
-// - its parent belongs to another service (in that case it's a "local root"
+// - OR its parent is unknown (other part of the code, distributed trace)
+// - OR its parent belongs to another service (in that case it's a "local root"
 //   being the highest ancestor of other spans belonging to this service and
 //   attached to it).
-func ComputeTopLevel(t pb.Trace) {
-	// build a lookup map
-	spanIDToIdx := make(map[uint64]int, len(t))
-	for i, span := range t {
-		spanIDToIdx[span.SpanID] = i
+func ComputeTopLevel(trace pb.Trace) {
+	spanIdToIndex := make(map[uint64]int, len(trace))
+	for i, span := range trace {
+		spanIdToIndex[span.SpanID] = i
 	}
 
-	// iterate on each span and mark them as top-level if relevant
-	for _, span := range t {
-		if span.ParentID != 0 {
-			if parentIdx, ok := spanIDToIdx[span.ParentID]; ok && t[parentIdx].Service == span.Service {
-				continue
-			}
+	for _, span := range trace {
+		isRootSpan := span.ParentID == 0
+		if isRootSpan {
+			SetTopLevel(span, true)
+			continue
+		}
+
+		parentIndex, isParentKnown := spanIdToIndex[span.ParentID]
+		if !isParentKnown {
```

Comment:
Same here. The convention in Go for checking if a map item exists is naming it `ok`
```suggestion
		parentIndex, ok := spanIDToIndex[span.ParentID]
		if !ok {
		   // has no parent in chunk
		   ...
		}
```